packages/nimble-components/src/rich-text/editor/tests/rich-text-editor-mention.spec.ts

```diff
@@ -1542,8 +1542,8 @@ describe('RichTextEditorMentionListbox', () => {
             expect(pageObject.isMentionListboxOpened()).toBeTrue();
         });
 
-        // Intermittent, see https://github.com/ni/nimble/issues/2219
-        xit('should show mention popup for multiple mention configuration elements', async () => {
+        // Firefox skipped, see https://github.com/ni/nimble/issues/2404
+        it('should show mention popup for multiple mention configuration elements #SkipFirefox', async () => {
```

Comment:
I don't think I understand the plan. What is going to drive understanding the intermittency in Firefox? The test won't be running and no logging is being added to gather more information on failure.
It seems like this will stay disabled in Firefox indefinitely. @jattasNI @aagash-ni

Reply:
A concrete proposal would be:
1) Take time to consider what should happen in a failure case and what extra information would be useful to gather from the page to help debug.
2) Re-enable it under the idea that it will not fail and if it fails again, restart the triage process. Seems undesirable because their seems to be knowledge that it is expected to fail intermittently on Firefox.
3) In this PR: https://github.com/ni/nimble/pull/2396/files#diff-4b859665afd3d8b0eab8e029e088d38102ae9bda4f89964d8efa3d718f6c397aR1379 it looks like you only ran the test in a loop 30 times. You could try pushing it harder. Mark the test as `fdescribe` to ignore other tests and maybe try 100, 1000, 10000 times? Thirty seems like a pretty low and oddly specific number to try and say that it's not reproducible.

Reply:
@aagash-ni I haven't seen any follow-up to thee review, moving the PR back to draft until can make further progress

Reply:
Hi @rajsite, We have tried a few more ways to find the root cause of the issue today. However, the cause has not yet been identified.

However, I am sharing some observations based on the above proposals:
1. After numerous attempts, the test for `should show mention popup for multiple mention configuration elements` failed only once due to the reason specified in the bug report. This occurred after running the test in a loop hundreds of times. Here is the error log.
     ![image](https://github.com/user-attachments/assets/54f4869a-62af-47b1-973d-6b2bb72607e7)

      The error was caused by `clickMentionListboxOption(0)` in the line above. However, for this test, committing the node into the editor was not necessary, and we were also not asserting the committed mention node. Instead, after asserting mention 1, I removed the `@` from the editor using the `sliceEditorContent` method from the page object. Then, added `!` and asserted the list box options and the visibility of the popup for other mention.
1. For the other test in the serializer, the test never failed even after many retries. However, based on the previous issue fix #2200, before committing the node from the listbox, focus the editor again by updating a page object in the `clickMentionListboxOption`.
      1. Can we try re-enabling with these changes and monitoring for a few more days? Maybe I will try to figure out any helpful logs that can be added.
1. In the other pull request (#2396), we attempted to focus on fixing the failing test and running it multiple times.
     1. Run both the tests 1000 times - No failures: https://github.com/ni/nimble/actions/runs/11160728028/job/31021797869?pr=2396#step:17:17743
     3. Run both the tests 10000 times - Pipeline exited with an error code: https://github.com/ni/nimble/actions/runs/11164365077/job/31033523031?pr=2396#step:17:472
     4. Run the tests 1000 times again with the changes explained in the above points (`#1` & `#2`) - the test in the editor failed 27 times (Only on Firefox) but for a different reason: https://github.com/ni/nimble/actions/runs/11161515146/job/31024221277?pr=2396#step:17:12176
           1. The failed test can only be reproduced if the test is in a loop and not in the ideal case. After further debugging, it was found that `isMentionListboxOpened` returned `false` because the `nimble-anchored-region` intermittently has the `hidden` attribute. This issue is not seen in the actual Firefox browser, but it is reproducible when running `npm run test-firefox:debugger` locally. So, could this be related to the Firefox launcher? Is debugging in that direction the right approach?

Please share your thoughts if you have any on these observations and let us know if we can proceed with the changes in the PR and monitor the fix for some more time.

Reply:
> Maybe I will try to figure out any helpful logs that can be added.

The only helpful log I could think of for this error is to get the focus status of the editor. I would like to understand if that's the case the test in the serializer is failing intermittently on the Firefox browser. The reason for this log is that the error (`No selection range exists`) can be thrown from the Tiptap, only if there are no selections (the editor might not be focused).

Therefore, can we add a log to the test in the PR? Or can we double-check by focusing on the editor again when `clickMentionListboxOption` is executed?

Reply:
@rajsite, Kindly review the changes in the PR and share your thoughts on the proposals and observations mentioned above.

Reply:
@rajsite Kindly review the changes in the PR and let us know your thoughts on the above proposals and observations.

Reply:
@vivinkrishna-ni I see you pushed the brainstorming ideas to the branch. To emphasize, I didn't prototype or test those ideas to see if they are actually beneficial or just hurt test performance by adding unnecessary waits.

It sounded like you had found a way to cause the failures somewhat reliably via many executive so hopefully the brainstorming ideas were tested against that

Reply:
@rajsite I've tried both of the suggestions (mentioned above) in the other PR (#2396) where we run the tests in a loop. It looks like the tests fail for the same reason. However, the frequency of failure is reduced, from 53 failures out of 2000 attempts to 37 failures out of 2000 attempts. In terms of property vs attribute and when logging the `hidden` property and attribute value, both are the same. So, I think adding the `await waitForUpdatesAsync();` before reading the `hidden` property/attribute makes the frequency difference I guess.

But, based on the other comment (https://github.com/ni/nimble/pull/2403#discussion_r1793940205) below, it fixes the issue. By focusing on the editor in the `sliceEditorContent` method, the issue has been fixed. I have updated the changes in the other PR, and the tests did not fail for all three attempts when running them in a loop for 2000 times each. Pipeline action: https://github.com/ni/nimble/actions/runs/11272830491/job/31351196076 (Verified the failures from the additional logs).

Therefore, I focused on the editor for all the methods in the page object that modify the content of the editor to avoid any other intermittencies. There are some lint failures in the PR which I will fix soon but apart from that please let me know whether can we try to fix this long-standing intermittent test failure with these changes. We also plan to monitor the failures for a few days before we close the bug.

Reply:
Thanks for testing in the other PR. Sounds very promising that the tests are passing more reliably! Generally looks good but can approve after the build passes 👍

Reply:
@rajsite, Please consider this a gentle reminder to review the PR and approve it if you don't have any comments.